data-raw/region_isos.R

```diff
@@ -0,0 +1,5 @@
+# Source: @jdhoffa
+path <- here::here("data-raw/region_isos.rds")
+region_isos <- readr::read_rds(path)
+
```

Comment:
@jdhoffa can you explain why a 2-column dataframe isn't a good alternative? 

A dataframe data structure is most common and easier to work with tidyverse tools. Sure, there is purrr, but not many people is fluent in list-oriented worflows. And it would be easy to extend `data_dictionary()`.

``` r
library(tidyverse)

region_isos_df <- r2dii.dataraw::region_isos %>% 
  purrr::map_df(as_tibble, .id = "region") %>% 
  dplyr::rename(isos = .data$value)

region_isos_df %>% 
  dplyr::group_by(region) %>% 
  # sampling a few rows per group to succinctly show some rows in several groups
  dplyr::filter(dplyr::row_number() %in% 1:3)
#> # A tibble: 72 x 2
#> # Groups:   region [26]
#>    region isos  
#>    <chr>  <chr> 
#>  1 global af    
#>  2 global al    
#>  3 global dz    
#>  4 africa dz    
#>  5 africa ao    
#>  6 africa bj    
#>  7 brazil br    
#>  8 brazil global
#>  9 china  cn    
#> 10 china  hk    
#> # ... with 62 more rows
```

<sup>Created on 2020-02-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

Reply:
Maybe what I'm missing is an example showing how you plan to use this dataset.

Reply:
And BTW, we are missing an R/regions_isos.R or R CMD check complaints about undocumented data.

Reply:
Ya so the question here is one of user-friendly vs. computer-friendly. 
To a user, hoping to define a new region, it is most intuitive to say: the region is called "foo" and in it are the countries c("bar1", "bar2", "bar3", ...) (easiest to define as a list i imagine). 
Then, in practice, I agree it would be easeier to have it as a 2-column dataframe. 

Question: If we paste the code you wrote above into the region_isos.R code in data-raw/, can we just have the package do the conversion on build? I will commit what I am talking about, and you can decide if this works or not.

Reply:
I think it's still easiest to just add a new list entry when we want to add a new region (in the data-raw/region_isos.rda list), but I have added a step in the data_prep function in data/raw to convert it to a 2-column dataframe afterward.

Reply:
Great. I now understand and I like that you are thinking not only about the user but also about the maintainers.

Reply:
i am both!